metals/src/main/scala/scala/meta/internal/decorations/SyntheticsDecorationProvider.scala

```diff
@@ -0,0 +1,300 @@
+    for {
+      focused <- focusedDocument()
+      filePath = AbsolutePath(path)
+      sourcePath <- SemanticdbClasspath.toScala(workspace, filePath)
+      if sourcePath == focused || !clientConfig.isDidFocusProvider()
+      textDoc <- textDocument.documents.headOption
+      source <- fingerprints.loadLastValid(sourcePath, textDoc.md5, charset)
+    } {
+      val docWithText = textDoc.withText(source)
+      Document.set(docWithText)
+      publishSyntheticDecorations(sourcePath, docWithText)
+    }
+  }
+
+  def addSyntheticsHover(
+      params: TextDocumentPositionParams,
+      pcHover: Option[l.Hover]
+  ): Option[l.Hover] = {
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    if (!clientConfig.isInlineDecorationProvider()) {
+      val line = params.getPosition().getLine()
+      currentDocument(path) match {
+        case Some(textDocument) =>
+          val edit = buffer.tokenEditDistance(path, textDocument.text)
+
+          val syntheticsAtLine = for {
+            synthetic <- textDocument.synthetics
+            range <- synthetic.range.toIterable
+            lspRange <- edit.toRevisedStrict(range).toIterable
+            if lspRange.getEnd.getLine == line
+            fullSnippet <- printSyntheticInfo(
+              textDocument,
+              synthetic,
+              toHoverString(textDocument),
+              userConfig(),
+              simple = false
+            ).toIterable
+          } yield (lspRange, fullSnippet)
+
+          if (syntheticsAtLine.size > 0) {
+            createHoverForLine(path, syntheticsAtLine, pcHover).orElse(
+              pcHover
+            )
+          } else {
+            pcHover
+          }
+        case None => pcHover
+      }
+
+    } else {
+      pcHover
+    }
+  }
+
+  private def createHoverForLine(
+      path: AbsolutePath,
+      syntheticsAtLine: Seq[(l.Range, String)],
+      pcHover: Option[l.Hover]
+  ): Option[l.Hover] =
+    Try {
+      val line = syntheticsAtLine.head._1.getEnd().getLine()
+      val newHover = for {
+        currentText <- buffer.get(path)
+        lineText <- currentText.linesIterator.drop(line).headOption
+      } yield {
+        val withEnd = syntheticsAtLine
+          .map {
+            case (range, str) => (range.getEnd().getCharacter(), str)
+          }
+          .sortBy(_._1) :+ (lineText.size, "")
+        val lineWithDecorations = withEnd
+          .foldLeft(("", 0)) {
+            case ((current, index), (char, text)) =>
+              val toAdd = lineText.substring(index, char) + text
+              (current + toAdd, char)
+          }
+          ._1
+
+        val previousContent = pcHover
+          .filter(_.getContents().isRight())
+          .map(_.getContents().getRight.getValue + "\n")
+          .getOrElse("")
+
+        new l.Hover(
+          new l.MarkupContent(
+            l.MarkupKind.MARKDOWN,
+            previousContent
+              + "**This line with synthetics added**:\n"
```

Comment:
What do you think of simply this?
```suggestion
              + "_**With synthetics added**_:\n"
```

Reply:
Actually hold off on this. I may be wrong.

From up above:

> I wonder why the markdown is not working there, seemed to work ok in VS Code. Any idea?

No need to worry about that, it's expected. Keep in mind that I'm in a terminal, so while some hacks are done in coc to make things bold etc, not everything will work as expected.

Also the bold in the italics doesn't work as expected, so don't do that. I'd just change the wording and keep the double `**`.

Reply:
Updated, thanks!